xhr/resources/inspect-headers.py

```diff
@@ -8,13 +8,14 @@ def get_response(raw_headers, filter_value, filter_name):
     # Python 3. <http.client.HTTPMessage> doesn't have 'headers" attribute or equivalent
     # [https://bugs.python.org/issue4773].
     # In Python 2, variable raw_headers.headers returns a completely uninterpreted list of lines
-    # contained in the header. In Python 3, raw_headers.as_string() returns entire formatted
-    # message as a string. Here is to construct an equivalent "headers" variable to support tests
-    # in Python 3.
+    # contained in the header. In Python 3, raw_headers.items() returns a list of 2-tuples
+    # containing all the message’s field headers and values. Here is to construct an equivalent
+    # "headers" variable to support tests in Python 3.
     if PY3:
-        header_list = [
-            isomorphic_encode(s + u'\r\n') for s in raw_headers.as_string().splitlines() if s
-        ]
+        header_list = []
+        for field in raw_headers.items():
```

Comment:
What does `items()` do though? How the lines are split is somewhat significant.

Reply:
`raw_items` is probably more direct, without additional parsing of header values (which `items` does). Unfortunately it's not documented as it's an "internal" but "public" API.

https://github.com/python/cpython/blob/8021875bbcf7385e651def51bc597472a569042c/Lib/email/message.py#L486

Either way, we don't have much control over how lines and key/value pairs are split. The http stdlib of Python 3 exposes even less raw info than that of Python 2, which is making the case of moving away from stdlib stronger, sigh...

Reply:
```suggestion
        for field in raw_headers.raw_items():
```